nemo_rl/models/generation/sglang/sglang_worker.py

```
@@ -0,0 +1,734 @@
+        with requests.Session() as session:
+            while True:
+                try:
+                    response = session.get(f"{self.base_url}/health_generate", headers=headers)
+                    if response.status_code == 200:
+                        print(f"[SGLang Server] Rank {self.global_rank} Server is ready at {self.base_url}")
+                        break
+                except requests.RequestException:
+                    pass
+
+                if not p.is_alive():
+                    raise Exception(f"[SGLang Server] Rank {self.global_rank} Server process terminated unexpectedly.")
```

Ruff (0.14.7)

Create your own exception

(TRY002)